### PR TITLE
Recreate security rules (ACL) when "clearing" GeoServer workspace

### PR DESCRIPTION
**Pull request by kannes**
A quick implementation of https://github.com/GeoCat/qgis-bridge-plugin/issues/169.

Upon workspace "clearing" (actually a deletion and creation of a new workspace) this:

1. Fetches existing ACL rules, e.g.:
```
existing_acl_rules={
  '*.*.r': '*',
  'test.*.r': 'ROLE_AUTHENTICATED',
  'test.*.w': 'ROLE_AUTHENTICATED,ROLE_LDAP_123',
  'foo.*.a': 'ROLE_OTHER',
  'foo.*.w': '*'
}
```

2. Extracts those that start with the workspace name and a dot (so that "foo2" won't be matched by "foo"), e.g.:
```
acl_rules={
  'test.*.r': 'ROLE_AUTHENTICATED',
  'test.*.w': 'ROLE_AUTHENTICATED,ROLE_LDAP_123'
}
```

3. Set those rules for the just re-created workspace.

-----

Fixes https://github.com/GeoCat/qgis-bridge-plugin/issues/169

### CODE REVIEW

**Comment by kannes** [comment]
Will force push a clean history later, without the commit about isolation.

File: geocatbridge/servers/models/geoserver.py [hunk 1]
```diff
@@ -1040,6 +1040,7 @@ def clearWorkspace(self, recreate=True) -> bool:
         # Get database datastores configuration and isolation flag
         db_stores = []
         isolated = False
+        acl_rules = {}
         if recreate:
             url = f"{self.apiUrl}/workspaces/{self.workspace}/datastores.json"
             stores = self.request(url).json()["dataStores"] or {}
```
Comment (GeoSander) [inline geocatbridge/servers/models/geoserver.py:1046]: Hi @kannes,

This has nothing to do with your work, but could you perhaps make this line a bit more robust while you are at it? Something like:

```suggestion
            stores = self.request(url).json().get("dataStores", {}) or {}
```

**Review by GeoSander — CHANGES_REQUESTED** [review]

**Review by GeoSander — CHANGES_REQUESTED** [review]
Not sure where that `or {}` was coming from, as it doesn't make sense how it was implemented here. 
However, if the `dataStores` property exists but it's `null`, `get("dataStores", {})` will return `None`. So adding `or {}` in that case is just an extra safety measure to avoid errors in the next call on line 1047.
I never trust the GeoServer REST API responses (they sometimes return an empty string or null in some JSON responses), so I do like to see that added here. Sorry :)

**Comment by kannes** [comment]
Ah, now I get it! Thanks for clarifying. I amended it to the last commit and force-pushed for a nice clean history :)

**Review by GeoSander — APPROVED** [review]
Looking good now, thanks @kannes!
Will merge.

**Pull request merged**